internal/component/redis/settings.go

```diff
@@ -32,6 +32,10 @@ type Settings struct {
 	DB int `mapstructure:"redisDB"`
 	// The redis type node or cluster
 	RedisType string `mapstructure:"redisType"`
+	// The redis version - controls which redis client to use
+	// >= 7 uses go-redis v9
+	// < 7 uses go-redis v8, this is also the default
+	RedisVersion int `mapstructure:"redisVersion"`
```

Comment:
Is there a way perhaps this could be determined at runtime? Maybe Dapr can ping the Redis server and determine its version? Not sure if that's even possible, just asking.

Reply:
Not from what I can tell.

Reply:
Are you able to give it a shot?

For example use the new SDK and if the version is old, reconnect with the old one. If it works, it would make the component much simpler as it wouldn't require operators to know the redis version beforehand

Reply:
Done and working.

Reply:
Noice!!!!